hazelcast/src/main/java/com/hazelcast/util/JVMUtil.java

```diff
@@ -36,12 +36,19 @@
      * Defines the costs for a reference in Bytes.
      */
     @SuppressWarnings("checkstyle:magicnumber")
-    public static final int REFERENCE_COST_IN_BYTES = isCompressedOops() ? 4 : 8;
+    public static final int REFERENCE_COST_IN_BYTES = is32bitJVM() || isCompressedOopsOrNull() ? 4 : 8;
```

Comment:
Yes, updated so that the name indicates that we use object layout to determine if compressed oops is enabled or not.